core/arch/mangle_shared.c

```diff
@@ -61,8 +61,8 @@ callee_info_t default_callee_info;
 int
 get_clean_call_switch_stack_size(void)
 {
-#ifdef AARCH64
-    /* Stack size needs to be 16 byte aligned on ARM */
+#if defined(AARCH64) || defined(X64)
+    /* Stack size needs to be 16 byte aligned on ARM and x64. */
```

Comment:
IMHO it would be cleaner to have a get_ABI_stack_alignment() function that returns a number.  WDYT?